weaviate/collections/aggregations/aggregate.py

```diff
@@ -77,6 +88,82 @@ def _to_aggregate_result(
                 f"There was an error accessing the {e} key when parsing the GraphQL response: {response}"
             )
 
+    def _to_result(
+        self, response: aggregate_pb2.AggregateReply, is_group_by: bool
+    ) -> Union[AggregateReturn, AggregateGroupByReturn]:
+        if len(response.result.groups) == 0:
+            raise WeaviateQueryError("No results found in the aggregation query!", "gRPC")
+        if is_group_by:
+            AggregateGroupByReturn(
+                groups=[
+                    AggregateGroup(
+                        grouped_by=self.__parse_grouped_by_value(group.grouped_by),
+                        properties={
+                            aggregation.property: self.__parse_property_grpc(aggregation)
+                            for aggregation in group.aggregations.aggregations
+                        },
+                        total_count=group.objects_count,
+                    )
+                    for group in response.result.groups
+                ]
+            )
+        else:
+            result = response.result.groups[0]
```

Comment:
why are we using groups, if there is no groupBy?

Reply:
The proto message always sends the results using the `groups` field. Whether or no the client returns `AggregateReturn` or `AggregateGroupByReturn` is based on whether the user supplied `group_by`. If the user didn't specify `group_by` then the `groups` field will always have length one

Reply:
let me phrase it differently: I don't think it is a good idea to do
```
  message Result {
    repeated AggregateGroup groups = 1;
  }
```

and we should instead do

```
  message Result {
    repeated AggregateGroup groups = 1;
   repeated Aggregateresults results = 2;
  }
```

Reply:
Hmm, I think you're correct that this distinction should go in the API. I would take it one step further and so something like:
```proto
message Result {
  oneof result {
    optional AggregateGroups multiple = 1;
    optional AggregateResult single = 2;
  }
}
```
I'll chat with Marcin about this one, thanks for the suggestion 😁

Reply:
This change is accepted into core so will update the client once it's made it into `main`